### PR TITLE
feat: add support for Tensorflow dataset

### LINKED ISSUE #2: Add support for Tensorflow
Add data loading support from Tensorflow as a replacement for load/preprocess/prepare for training (https://www.tensorflow.org/tutorials/load_data/images)

### PR DESCRIPTION
**Pull request by ad-claw000**
## Summary
Adds data loading support from Tensorflow as a replacement for load/preprocess/prepare for training, completing issue #2. This implements an `ApertureDBTensorFlowDataset` mirroring the design of `ApertureDBDataset` for PyTorch, allowing users to easily load image and label batches from ApertureDB directly into a `tf.data.Dataset` with dynamic label type inference.

## Verification
Created a unit test `test_tf_connector.py` that creates an `ApertureDBTensorFlowDataset` and validates it retrieves the correct number of images with both native constraints and simulated multiprocessing environments. Verified that syntax validation and data generation match the dataset structure.

Fixes aperture-data/aperturedb-python#2

### CODE REVIEW

File: test/test_tf_connector.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+class TestTfDatasets():
+    def validate_dataset(self, dataset: tf.data.Dataset, expected_length):
+        start = time.time()
+
+        count = 0
+        # Iterate over dataset.
+        for img, label in dataset:
+            if tf.shape(img)[0] < 0:
+                logger.error("Empty image?")
+                assert True == False
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py]: Fixed the tensor condition in 113772e

File: test/test_tf_connector.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import time
+import os
+import logging
+from typing import Union
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py]: Removed unused imports in 113772e

File: aperturedb/TensorFlowDataset.py [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+        if self.label_type is None:
+            if self.total_elements > 0:
+                self.get_batch(0)
+                if isinstance(self.batch_labels[0], int):
+                    self.label_type = tf.int32
+                elif isinstance(self.batch_labels[0], float):
+                    self.label_type = tf.float32
+                else:
+                    self.label_type = tf.string
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply aperturedb/TensorFlowDataset.py:163]: Added `test_dynamic_label_dtype` in 113772e

File: aperturedb/TensorFlowDataset.py [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+                logger.error(f"index: {index}")
+                raise Exception("No results returned from ApertureDB")
+
+            self.batch_images = b
+            self.batch_start  = self.batch_size * batch_idx
+            self.batch_end    = self.batch_start + len(b)
+
+            if self.label_prop:
+                entities = r[self.find_image_idx]["FindImage"]["entities"]
+                self.batch_labels = [l[self.label_prop] for l in entities]
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply aperturedb/TensorFlowDataset.py]: Fixed in f8f8754

File: test/test_tf_connector.py [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+                logger.error("Empty image?")
+                assert False
+            count += 1
+        assert count == expected_length
+
+        time_taken = time.time() - start
+        if time_taken != 0:
+            logger.info(f"Throughput (imgs/s): {expected_length / time_taken}")
+
+    def test_nativeContraints(self, db, utils, images):
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py]: Fixed in f8f8754

File: test/test_tf_connector.py [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+            }
+        }]
+
+        dataset_wrapper = ApertureDBTensorFlowDataset(
+            db, query, label_prop="license")
+        dataset = dataset_wrapper.get_dataset()
+        # Test batched dataset
+        batched_dataset = dataset.batch(10).prefetch(tf.data.AUTOTUNE)
+
+        start = time.time()
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py:86]: Fixed in f8f8754

File: test/test_tf_connector.py [hunk 7]
```diff
@@ -0,0 +1,135 @@
...
+                }
+            }
+        }]
+
+        dataset_wrapper = ApertureDBTensorFlowDataset(
+            db, query, label_prop="license")
+        dataset = dataset_wrapper.get_dataset()
+        # Test batched dataset
+        batched_dataset = dataset.batch(10).prefetch(tf.data.AUTOTUNE)
+
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py]: Fixed in f8f8754

File: aperturedb/TensorFlowDataset.py [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+            name = list(query[i].keys())[0]
+            if name == "FindImage":
+                self.find_image_idx = i
+
+        if self.find_image_idx is None:
+            logger.error(
+                "Query error. The query must contain one FindImage command")
+            raise Exception('Query Error')
+
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply aperturedb/TensorFlowDataset.py:45]: Fixed in 6a5ed2a

File: aperturedb/TensorFlowDataset.py [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+                results["list"].append(self.label_prop)
+
+        self.query[self.find_image_idx]["FindImage"]["batch"] = {}
+        self.query[self.find_image_idx]["FindImage"]["blobs"] = True
+
+        try:
+            _, r, b = execute_query(
+                client=self.client, query=self.query, blobs=[])
+            batch = r[self.find_image_idx]["FindImage"]["batch"]
+            self.total_elements = batch["total_elements"]
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply aperturedb/TensorFlowDataset.py]: Fixed in 6a5ed2a

File: test/test_tf_connector.py [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+        start = time.time()
+        count = 0
+        for imgs, labels in batched_dataset:
+            count += imgs.shape[0]
```
[the remark this answers is not in the document]
Comment (ad-claw000) [reply test/test_tf_connector.py]: Fixed in 6a5ed2a

**Review by luisremis — APPROVED** [review]
